### DISCUSSION TITLE
How to stop Prisma from overwriting custom model/field names?

### DISCUSSION THREAD

**Discussion by Inayat-C** [prisma/prisma#2655] [Q&A] [answered]
I'm trying to adhere to the Prisma naming conventions, for example by making sure my field names are spelt in camelcase. However, whenever i add a new table in my database and run `npx prisma introspection`, i expectedly lose my edits as the schema file is overwritten. 

Is there anyway i could work around this where i'm able to keep the custom edits i've done to the schema file? Otherwise i don't see any point in putting effort to make sure my models/fields are named appropriately. 

Thanks!

**Comment by ryands17** [comment, marked as answer]
Hey @Inayat-C :wave: 
I had posted this snippet https://github.com/prisma/prisma/discussions/2530#discussioncomment-17535 regarding how to model the names after introspect.

So this script can automatically be set to run after introspect runs.

Let me know if this works for you :)

**Reply by Inayat-C** [reply]
Thanks @ryands17, it seems to be working on my very small schema file. Would be curious to see how it performs once the schema grows. 

By the way, do you think there would be any issue if i chain introspect, fixPrismaFile and generate to run automatically in that order on one command?

**Reply by ryands17** [reply]
@Inayat-C no there wouldn't be any issue if you chain those together. I have done that myself as follows:

```json
"introspect": "yarn prisma introspect",
"postintrospect": "yarn fixmodels && yarn prisma generate"
```

**Reply by Inayat-C** [reply]
Awesome, i'll give it a go, thanks!

**Reply by ryands17** [reply]
Do let me know how it goes and if you're trying with large models as well :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
